src/programs/BuddyProject.ts

```diff
@@ -59,44 +65,48 @@ export async function BuddyProjectSignup(
         where: { discord_user: !discord_user, matched: false },
       });
       if (potentialMatches.length > 0) {
-        await buddyEntries
-          .findOne({ where: { discord_user: !discord_user } })
-          .then((res) => {
-            updateDatabaseWithQuery(
-              buddyEntries,
-              member.id,
-              res.user_id,
-              BuddyEntry
-            );
-
-            new Client().users.fetch(res.buddy_id).then((mm) =>
-              mm.createDM().then((buddyDm) => {
-                buddyDm.send("Here is your match: " + member.id);
-              })
-            );
-            dmChannel.send("Here is your match:" + res.user_id);
-          });
-        return null;
-      }
-
-      // If the opposite group didn't have any matches, find one from own group
-      await buddyEntries
-        .findOne({ where: { discord_user: discord_user, matched: false } })
-        .then((res) => {
+        try {
+          const finalMatch = potentialMatches[0];
           updateDatabaseWithQuery(
             buddyEntries,
             member.id,
-            res.user_id,
+            finalMatch.user_id,
             BuddyEntry
           );
-
-          new Client().users.fetch(res.buddy_id).then((mm) =>
-            mm.createDM().then((buddyDm) => {
-              buddyDm.send("Here is your match: " + "<@" + member.id + ">");
-            })
+          const buddyDmClient = await new Client().users.fetch(
```

Comment:
You can `import bot from "../index";` to avoid having to create a new one each time someone signs up for the buddy project.

```suggestion
          const buddyDmClient = bot.users.fetch(
```